File: cpp/org.kevoree.modeling.cpp.microframework/src/main/cpp/microframework/api/utils/ModelVisitor.cpp
#include <microframework/api/utils/ModelVisitor.h>

void ModelVisitor::stopVisit()
{
	visitStopped = true;
}

void ModelVisitor::noChildrenVisit()
{
	visitChildren = true;
}


ModelVisitor::ModelVisitor(){
	visitStopped = false;
	visitChildren = true;
	visitReferences=true;
}


void ModelVisitor::noReferencesVisit(){
	visitReferences = false;
}

File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/ModelVisitor.kt
package org.kevoree.modeling.api

import java.util.HashMap

public abstract class ModelVisitor {

    var visitStopped = false

    fun stopVisit() {
        visitStopped = true
    }

    var visitChildren = true

    var visitReferences = true

    fun noChildrenVisit() {
        visitChildren = false
    }

    fun noReferencesVisit() {
        visitReferences = false
    }

    public abstract fun visit(elem: KObject<*>, refNameInParent: String, parent: KObject<*>)

    var alreadyVisited: HashMap<String, KObject<*>>? = null

    open fun beginVisitElem(elem: KObject<*>) {
    }

    open fun endVisitElem(elem: KObject<*>) {
    }

    open fun beginVisitRef(refName: String, refType: String): Boolean {
        return true;
    }

    open fun endVisitRef(refName: String) {
    }

}